try catch trycatch java io exception ioexception test method reference testmethodreference test test io exception ioexception foo